cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -362,26 +362,68 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 	return nil
 }
 
-func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
+// getTagsMap extract the key-value pairs from openapi tags and struct tags returns a map[string]string
+func getTagsMap(CommentLines []string, tags string) (map[string]string, error) {
+	tagMap := make(map[string]string)
+
+	// Handle openapi tags
 	tagValues := getOpenAPITagValue(CommentLines)
-	anyExtension := false
 	for _, val := range tagValues {
 		if strings.HasPrefix(val, tagExtensionPrefix) {
-			if !anyExtension {
-				g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
-				anyExtension = true
-			}
 			parts := strings.SplitN(val, ":", 2)
 			if len(parts) != 2 {
-				return fmt.Errorf("Invalid extension value: %v", val)
+				return nil, fmt.Errorf("Invalid extension value: %v", val)
 			}
-			g.Do("\"$.$\": ", parts[0])
-			g.Do("\"$.$\",\n", parts[1])
+			tagMap[parts[0]] = parts[1]
 		}
 	}
-	if anyExtension {
-		g.Do("},\n},\n", nil)
+
+	// Handle struct tags
+	tagPairs := strings.Fields(tags)
+	for _, tagPair := range tagPairs {
+		parts := strings.SplitN(tagPair, ":", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("Invalid extension value: %v", tagPair)
+		}
+		switch parts[0] {
+		case "patchStrategy":
```

Comment:
I see only 92 matches for `patchStrategy` and 83 matches for `patchMergeKey` in main repo (the ones that are in string literal, I assume those are the only one we care about, also I guess both cases have near 100% overlap). Duplicating them to comment tag should not take much time. Also instead of validating during openapi generation, why don't we just add a test using gengo. It shouldn't be hard to do. Right now openapi code is really generic and I would much prefer to keep it that way. You are right about `x-kubernetes` though. That, however, is a constant that can be easily convert to a setting and moved out of here.